tests/State/HubAdapterTest.php

```diff
@@ -47,6 +47,9 @@ public function testGetInstanceReturnsUnserializableInstance(): void
         $this->expectException(\BadMethodCallException::class);
         $this->expectExceptionMessage('Unserializing instances of this class is forbidden.');
 
+        $this->expectException(\BadMethodCallException::class);
```

Comment:
You should split the test into two, because only one exception can be thrown at time. Furthermore, `expectException()` is in no case cumulative. I suggest to rename this test method to `testHubAdapterThrowsExceptionOnSerialization`, and add `testHubAdapterThrowsExceptionOnSerialization`

Reply:
How should we test `unserialize` when we need to call `serialize` first?

Reply:
We could hardcode the argument passed to `unserialize()` to be `O:10:"HubAdapter":0:{}`